decidim-core/lib/decidim/content_security_policy.rb

```diff
@@ -0,0 +1,121 @@
+      child-src
+      connect-src
+      default-src
+      font-src
+      frame-src
+      img-src
+      manifest-src
+      media-src
+      object-src
+      prefetch-src
+      script-src
+      script-src-elem
+      script-src-attr
+      style-src-elem
+      style-src-attr
+      worker-src
+      base-uri
+      sandbox
+      form-action
+      frame-ancestors
+      navigate-to
+      report-uri
+      report-to
+      require-trusted-types-for
+      trusted-types
+      upgrade-insecure-requests
+      style-src
+    ).freeze
+
+    def initialize(organization = nil, additional_policies = {})
+      @organization = organization
+      @policy = default_policy
+      @additional_policies = additional_policies
+    end
+
+    def output_policy
+      add_system_csp_directives
+      add_additional_policies
+      organization_csp_directives
+
+      format_policies
+    end
+
+    def append_csp_directive(directive, value)
+      return if value.blank?
+
+      message = "Invalid Content Security Policy directive: #{directive}, supported directives: #{SUPPORTED_POLICIES.join(", ")}"
+      raise message unless SUPPORTED_POLICIES.include?(directive)
+
+      policy[directive] ||= []
+      policy[directive] << value
+    end
+
+    private
+
+    attr_reader :policy, :organization, :additional_policies
+
+    def format_policies
+      policy.map do |directive, values|
+        [directive, values.uniq.join(" ")].join(" ")
+      end.join("; ")
+    end
+
+    def append_multiple_csp_directives(directive, values)
+      values.each do |v|
+        append_csp_directive(directive, v)
+      end
+    end
+
+    def add_additional_policies
+      additional_policies.each do |directive, values|
+        next if values.blank?
+
+        values = values.split if values.is_a?(String)
+
+        raise "Invalid value for additional CSP policy #{directive}: #{values.inspect}" unless values.is_a?(Array)
+
+        append_multiple_csp_directives(directive, values)
+      end
+    end
+
+    def organization_csp_directives
+      return unless organization
+
+      organization.content_security_policy.each do |directive, value|
+        append_multiple_csp_directives(directive, value.split) if value.present?
+      end
+    end
+
+    def add_system_csp_directives
+      if Rails.configuration.action_controller.asset_host
+        %w(media-src img-src script-src style-src).each do |directive|
+          append_csp_directive(directive, Rails.configuration.action_controller.asset_host)
+        end
+      end
+    end
+
+    # rubocop:disable Lint/PercentStringArray
+    def default_policy
+      {
+        "default-src" => %w('self' 'unsafe-inline'),
+        "script-src" => %w('self' 'unsafe-inline' 'unsafe-eval'),
+        "style-src" => %w('self' 'unsafe-inline'),
+        "img-src" => %w('self' *.hereapi.com data:),
+        "font-src" => %w('self'),
+        "connect-src" => %w('self' *.hereapi.com *.jsdelivr.net),
```

Comment:
To be honest, i am running on other ports (docker config). I need to find another way of approaching this.

Reply:
> Why are we allowing jsdelivr.net? Is there any script/library that needs this?

removing jsdeliver would cause the suite to fail with the message: 
```
2023-06-11 12:13:36 +0000 SEVERE: http://62.lvh.me:5535/packs-test/js/22-1100311c558b11060e03.js 1 Refused to connect to 'https://cdn.jsdelivr.net/npm/emojibase-data@latest/en/data.json' because it violates the following Content Security Policy directive: "connect-src 'self' *.hereapi.com".
2023-06-11 12:13:36 +0000 SEVERE: http://62.lvh.me:5535/packs-test/js/22-1100311c558b11060e03.js 1 Refused to connect to 'https://cdn.jsdelivr.net/npm/emojibase-data@latest/en/data.json' because it violates the document's Content Security Policy.
2023-06-11 12:13:36 +0000 SEVERE: http://62.lvh.me:5535/packs-test/js/22-1100311c558b11060e03.js 1 Refused to connect to 'https://cdn.jsdelivr.net/npm/emojibase-data@latest/en/messages.json' because it violates the following Content Security Policy directive: "connect-src 'self' *.hereapi.com".
```